casual/casual-caller-http-client/src/test/groovy/se/laz/casual/http/HttpClientTest.groovy

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (c) 2024, The casual project. All rights reserved.
+ *
+ * This software is licensed under the MIT license, https://opensource.org/licenses/MIT
+ */
+
+package se.laz.casual.http
+
+import jakarta.ws.rs.core.Application
+import org.glassfish.jersey.server.ResourceConfig
+import org.glassfish.jersey.test.JerseyTest
+import se.laz.casual.api.buffer.CasualBuffer
+import se.laz.casual.api.buffer.ServiceReturn
+import se.laz.casual.api.buffer.type.CStringBuffer
+import se.laz.casual.api.buffer.type.JsonBuffer
+import se.laz.casual.api.buffer.type.OctetBuffer
+import se.laz.casual.api.buffer.type.fielded.FieldedTypeBuffer
+import se.laz.casual.api.flags.ErrorState
+import se.laz.casual.api.flags.ServiceReturnState
+import se.laz.casual.http.test.resource.TestPath
+import se.laz.casual.http.test.resource.TestResource
+import spock.lang.Shared
+import spock.lang.Specification
+
+import java.nio.charset.StandardCharsets
+
+class HttpClientTest extends Specification
+{
+   @Shared
+   HttpClient client = HttpClient.of()
+   @Shared
+   def content = '{"msg":"bazinga!"}'
+   @Shared
+   def key = 'FLD_STRING1'
+   @Shared
+   def root = 'test'
+
+   JerseyTest resource = new JerseyTest(){
```

Comment:
This seems like a lot of work to create a whole server implementation just to test the client.
Why not utilise RestClientDriver instead?
https://github.com/rest-driver/rest-driver

Just declare the rule:
```groovy
@Rule
public ClientDriverRule driver = new ClientDriverRule()
```
Then add your expectation in your `given`:
```groovy
        driver.addExpectation(
                onRequestTo( "/casual/servicename1" )
                        .withAnyParams()
                        .withBody( expectedJsonPayload, MediaType.APPLICATION_JSON )
                        .withMethod( ClientDriverRequest.Method.POST ),
                giveResponse( fakeJsonResponse, MediaType.APPLICATION_JSON )
        )
```
This uses assertions that the client actually invokes the expected http request with the expected data.
It then allows you to return the response data you wish or error codes, which can easliy be use with a `where` table:

```groovy
driver.addExpectation(
                onRequestTo( "/casual/servicename1"T )
                        .withAnyParams()
                        .withBody( toHalJson( workspace ), MediaType.APPLICATION_JSON )
                        .withMethod( ClientDriverRequest.Method.POST ),
                giveEmptyResponse().withStatus( status )
        )
```
It can also flag up if other http requests are made that are not expected.

Reply:
Curious, why withAnyParams, they do not matter?
Usually you are extremely keen on matching parameters but not here?

Sure I can rewrite it usint ClientDriverRule, however - that would be a lot more work than letting it be as it is.

Reply:
Yes in mocks I don't like wildcards. These examples were just a quick copy paste from elsewhere, that I don't care what params are sent on a http level as they are not used at all. There are also methods for `withHeader`, `withoutHeader`. Use what is needed.

The reason I recommend this approach over the other is that it confirms the entire http interaction including the payload sent on the wire, which is not checked currently. 
If you don't call it with the right payload it will raise an exception during the test.

It shouldn't need to be that different total to the code you have now but it will improve readability.